elasticsearch_collectd.py

```diff
@@ -708,27 +1074,53 @@ def load_es_info():
 def parse_node_stats(json, stats):
     """Parse node stats response from ElasticSearch"""
     for name, key in stats.iteritems():
-        result = lookup_node_stat(name, json)
-        dispatch_stat(result, name, key)
+        if DETAILED_METRICS is True or name in DEFAULTS:
+            result = lookup_node_stat(name, json)
+            dispatch_stat(result, name, key)
+
+
+def parse_thread_pool_stats(json, stats):
+    """Parse thread pool stats response from ElasticSearch"""
+    for pool in THREAD_POOLS:
+        for metric_type, value in THREAD_POOL_METRICS.iteritems():
+            for attr in value:
+                name = 'thread_pool.{0}'.format(attr)
+                key = Stat(metric_type, 'nodes.%s.thread_pool.{0}.{1}'.
+                           format(pool, attr))
+                if DETAILED_METRICS is True or name in DEFAULTS:
+                    node = json['nodes'].keys()[0]
+                    result = dig_it_up(json, key.path % node)
+                    # Check to make sure we have a valid result
+                    # dig_it_up returns False if no match found
+                    if not isinstance(result, bool):
+                        result = int(result)
+                    else:
+                        result = None
```

Comment:
This borrows heavily from parse_node_stats() -> lookup_node_stat() -> dispatch_stat().  In that flow there isn't any logging for errors until the dispatch_stat() function.  I am staying consistent with the other functions that invoke dispatch_stat().